src/lib/web.py

```diff
@@ -0,0 +1,349 @@
+        self.wait_switch_frame(By.ID, "ide-application-iframe")
+        # Wait for all the page frames to load
+        time.sleep(30)
+        # Switch to the last frame to find main html content for configuration
+        self.wait_switch_frame(By.TAG_NAME, "iframe")
+        self.wait_switch_frame(By.XPATH, "//iframe[@id='active-frame']", 60)
+        self.wait_switch_frame(By.TAG_NAME, "iframe")
+        # Enter project name and select quarkus 1 as target
+        self.wait_click_element(By.XPATH, "//span[@id='input-details']/dl/dd/div/a")
+        self.write_text(
+            By.XPATH, "//input[@id='editDialogInput']", config_data["workspace_project_path"],
+        )
+        self.driver.find_element_by_xpath("//input[@id='editDialogInput']").send_keys(Keys.ENTER)
+        self.wait_click_element(By.XPATH, "//input[@id='target-quarkus1']")
+        # Get the configuration name and locator from sidebar
+        self.driver.switch_to.default_content()
+        self.wait_switch_frame(By.ID, "ide-application-iframe")
+        run_configurations = self.driver.find_elements_by_xpath(
+            "//div[@class='ReactVirtualized__Grid__innerScrollContainer' and @role='rowgroup']/div",
+        )
+        total_configs = len(run_configurations)
+        config_to_run = self.wait_find_element(
+            By.XPATH,
+            "//div[@class='ReactVirtualized__Grid__innerScrollContainer' \
+            and @role='rowgroup']/div[{}]".format(
+                total_configs,
+            ),
+        )
+        config_name_ele = self.wait_find_element(
+            By.XPATH,
+            "//div[@class='ReactVirtualized__Grid__innerScrollContainer' and \
+                @role='rowgroup']/div[{}]//div[@class='noWrapInfoTree']/span".format(
+                total_configs,
+            ),
+        )
+        configuration_name = str(config_name_ele.text)
+        action = ActionChains(self.driver)
+        # Right click action and run the analysis
+        action.context_click(config_to_run).perform()
+        self.wait_click_element(
+            By.XPATH, "//div[@class='p-Widget p-Menu']/ul/li/div[contains(text(), 'Run')]",
+        )
+        # Wait for analysis to start
+        self.wait_find_element(
+            By.XPATH,
+            "//div[@class='theia-notifications-container theia-notification-toasts open']\
+                //span[contains(text(), 'Analysis in progress')]",
+            60,
+        )
+        # Wait for analysis to complete
+        self.wait_find_element(
+            By.XPATH,
+            "//div[@class='theia-notifications-container theia-notification-toasts open']\
+                //span[contains(text(), 'Analysis complete')]",
+            120,
+        )
+        # Open report
+        self.wait_click_element(
+            By.XPATH,
+            "//div[@class='theia-notifications-container theia-notification-toasts open']\
+                //button[contains(text(), 'Open Report')]",
+        )
+        time.sleep(3)
```

Comment:
It will be better to use wait_for_element here .

Reply:
There is a tab switch at this moment, it happens so fast, that it tries to search for next tab element in previous one. Hence, this sleep

Reply:
@nitishSr :One more thought ,  can we add this wait time to wait_click_element method so that it always waits 10s . Will it make the automation slow?